For next

PeterMacleodThompson opened the pull request:
Update of ozmaps/mapmanager.c - 4 commits:
ozmaps: mapmanager.c: add 4 function prototypes to fix compiler warnings
ozmaps: Remove DEBUG define from mapmanager.c
ozmaps: mapmanager.c end getmap() function properly to remove compiler warnings
ozmaps: mapmanager.c make comments readable where clang-format made unreadable

This fixes mapmanager.c except for 1 compiler error which will be fixed in a later commit.

UPDATE -  Feb 8: 4pm
Another 9 commits were added here to:
 - finish removing DEBUG define
 - removes all compiler errors!!
 - add/change some comments
 - cleanup long descriptive comments which were muddled by clang-format

TODO 
3 new branches (I hope) to keep separate from the above commits:
 - dieface  with 2 commits to prepare for integration of dieface/viewsensor.c into hikingGPS  
       The daemons will remain separate of course. 
 - BBB with 2-3 commits to distinguish beaglebone black compiles via -D BBB
 - SDL2init with 3 commits to 
      replace a deprecated SDL2 command, 
      print SDL2 version,
      move find_datadir() from displaymap.c to SDL2init.c

PeterMacleodThompson committed:
ozmaps: mapmanager.c: add 4 function prototypes to fix compiler warnings

Function prototypes (4 of them)  added to ozmaps/mapmanager.c:
int keyadd(int, int);
int keysubtract(int, int);
int gpstogpskey(uint64_t);
int inthemap(int, PMTmapindex *);

to remove compiler warnings:
warning: implicit declaration of function ‘keyadd’
warning: implicit declaration of function ‘gpstogpskey’
warning: implicit declaration of function ‘gpstogpskey’;
warning: implicit declaration of function ‘inthemap’;

Signed-off-by: Peter Macleod Thompson <peter.macleod.thompson@gmail.com>

PeterMacleodThompson committed:
ozmaps: Remove DEBUG define from mapmanager.c

Initially many oz programs were created with a main() routine built
 within the program for unit testing.
The #define DEBUG was used to compile the main() routine
 (or not compile it if commented out).
The #define DEBUG concept along with main() routines for unit testing
 were removed from the oz system in an earlier commit
"76e0f7e Remove all main functions under DEBUG define".
This is a cleanup from that commit.

A #define DEBUG statement was removed.
A #ifdef DEBUG codeblock containing SDL2 include headers
 and SCREEN_WIDTH was removed.

Signed-off-by: Peter Macleod Thompson <peter.macleod.thompson@gmail.com>

PeterMacleodThompson committed:
ozmaps: mapmanager.c: end getmap() function properly to remove compiler warning

At the end of function ‘getmap’ in mapmanager.c
add the lines
  } else
     return (mapNow);  /* should never reach here */
so as to remove the compiler warning
warning: control reaches end of non-void function

Signed-off-by: Peter Macleod Thompson <peter.macleod.thompson@gmail.com>

PeterMacleodThompson committed:
ozmaps: mapmanager.c: make comments readable where clang-format made unreadable

When reformating all source code using clang-format, many long
descriptive comments became unreadable.
This is a manual fixup of the descriptive comments in mapmanager.c

PeterMacleodThompson committed:
ozmaps: mapmanager.c: clang-format changes to mapmanager.c

Forgot to clang-format the changes to mapmanager.c in last commit.

Signed-off-by: Peter Macleod Thompson <peter.macleod.thompson@gmail.com>

PeterMacleodThompson committed:
ozmaps: mapmisc.c Remove DEBUG define

Initially many oz programs were created with a main() routine built
 within the program for unit testing.
The #define DEBUG was used to compile the main() routine
 (or not compile it if commented out).
The #define DEBUG concept along with main() routines for unit testing
 were removed from the oz system in an earlier commit
"76e0f7e Remove all main functions under DEBUG define".
This is a cleanup from that commit.

A #define DEBUG statement was removed.
A #ifdef DEBUG codeblock of printf statements removed.

Signed-off-by: Peter Macleod Thompson <peter.macleod.thompson@gmail.com>

PeterMacleodThompson committed:
ozmaps: mapmisc.c: add FIXME comment re long/lat structure

The long/lat structure maybe should change but changing is a big job.
The added comment makes sure this is not forgotten.

Signed-off-by: Peter Macleod Thompson <peter.macleod.thompson@gmail.com>

PeterMacleodThompson committed:
ozdisplay: Remove DEBUG define displaymap.c, SDL2init.c SDL2misc.c

Initially many oz programs were created with a main() routine built
 within the program for unit testing.
The #define DEBUG was used to compile the main() routine
 (or not compile it if commented out).
The #define DEBUG concept along with main() routines for unit testing
 were removed from the oz system in an earlier commit
"76e0f7e Remove all main functions under DEBUG define".
This is a cleanup from that commit.

A #define DEBUG statement was removed from
displaymap.c
SDL2init.c
SDL2misc.c
A #ifdef codeblock was removed in SDL2misc.c

added comments to SDL2misc.c to better explain sprite structure

Signed-off-by: Peter Macleod Thompson <peter.macleod.thompson@gmail.com>

PeterMacleodThompson committed:
ozdisplay: displaymap.c: Remove unused variables mymapindex, gps, flip

Fixed compiler warnings for unused variables in ozdisplay/displaymap.c
warning: unused variable ‘mymapindex’
warning: unused variable ‘gps’
warning: unused variable ‘flip’

Signed-off-by: Peter Macleod Thompson <peter.macleod.thompson@gmail.com>

PeterMacleodThompson committed:
ozdisplay: displaymap.c: change main() to int main()

compiler warning: return type defaults to ‘int’ [-Wimplicit-int]
 main()
This is because main() ends with return 0; statement
fixed by changing main() to int main()

Signed-off-by: Peter Macleod Thompson <peter.macleod.thompson@gmail.com>

PeterMacleodThompson committed:
ozdisplay: displaymap.c: add a function prototype to fix compiler warnings

Function prototype
void closemaps(PMTmapset *);
 added to ozdisplay/displaymap.c
to fix compiler warning:
warning: implicit declaration of function ‘closemaps’;

Signed-off-by: Peter Macleod Thompson <peter.macleod.thompson@gmail.com>

PeterMacleodThompson committed:
include: maps.h  make comments readable where clang-format made unreadable

When reformating all source code using clang-format, many long
descriptive comments became unreadable.
This is a manual fixup of the descriptive comments in maps.h

Signed-off-by: Peter Macleod Thompson <peter.macleod.thompson@gmail.com>

PeterMacleodThompson committed:
ozmaps: mapmanager.c: remove %llu compiler error

compiler error = warning: format ‘%llu’ expects argument of type
 ‘long long unsigned int’, but argument 2 has type ‘uint64_t
resolved by typecasting printf variable to (long long) before printing
Since long long has different meanings on 32 bit vs 64 bit machines
 typecasting is easiest solution for portability.

Signed-off-by: Peter Macleod Thompson <peter.macleod.thompson@gmail.com>

eballetbo commented on src/ozmaps/mapmanager.c:
```diff
@@ -60,16 +60,6 @@ int keysubtract(int, int);
-#define SCREEN_HEIGHT 500
-
-#endif
 
```
note: If you follow the clang-format you should also remove one line, so there is only one line between the end of inthemap function and the comment.

eballetbo commented on src/ozmaps/mapmanager.c:
```diff
@@ -248,7 +248,8 @@ SDL_Surface *getmap(int mapflags,      /* instructions on which map to get */
-  }
+  } else 
+     return (mapNow);  /* should never reach here */
 }
```
You didn't run `clang-format -i` after do the modifications.

eballetbo replied:
Yeah, you should really fix the format in the commit that introduced the problem, not create a new commit to fix the format.

eballetbo commented on src/ozmaps/mapmisc.c:
```diff
@@ -109,10 +107,6 @@ int gpstogpskey(uint64_t gps) {
-#endif
 
   return (gpskey);
 }
```
lgtm

eballetbo commented on src/ozdisplay/SDL2misc.c:
```diff
@@ -45,23 +43,7 @@ process...
-#endif
-
+/*** load spritesheet from file ***/
 /* FIXME memory leak!!! sprite never gets SDL_FreeSurface( sprite ); */
```
That change is unrelated to the commit

eballetbo commented on src/ozdisplay/SDL2misc.c:
```diff
@@ -77,7 +59,9 @@ SDL_Surface *getsprite(char *filename) {
+ * can be any size (pixels) so long as it is square
+ * number of animations within sprite computed via sprite width/height
+ * returns number of animations. Completes array of SDL_Rects for extraction
  */
```
That change is unrelated to the commit

PeterMacleodThompson replied:
Yes - it maybe should have been a separate commit.    My lame excuse is that while inside the code, I couldn't remember how sprites worked, and was disgusted with my poor description so after significant work to re-figure sprites out, I changed the comments immediately.   The change was noted in the details of the  commit, but not the header....

eballetbo commented on src/ozdisplay/displaymap.c:
```diff
@@ -173,8 +171,6 @@ main() {
-  int mymapindex; /* %100 = mapsetindex, /100 = mapindex within mapset*/
 
   /******************** Step 1 game initialize **************************/
 
```
lgtm

eballetbo commented on src/ozdisplay/displaymap.c:
```diff
@@ -112,7 +112,7 @@ static char *find_datadir() {
+int main() {
   SDL_Surface *mymap; /*mymap= active map on screen */
   SDL_Surface *spritesheet;
   SDL_Surface *buttonsheet;
```
lgtm

eballetbo commented on src/ozdisplay/displaymap.c:
```diff
@@ -74,6 +74,7 @@ int scalepoint(SDL_Rect *, SDL_Point *, SDL_Point *);
+void closemaps(PMTmapset *);
 
 char *datapath = NULL;
 
```
lgtm

eballetbo reviewed (changes requested):
Hi Peter,

You're doing really well, I did some trivial comments, that again, because I'm strict for now, I'd like you to address, but basically is that you forgot to run clang-format -i before committing the changes.

Can we leave the documentation modifications that you did out from this pull request for now? I'd like to propose a documentation format instead.

So try for now only push the fixes for the warnings.

PeterMacleodThompson commented:
> Hi Peter,
> 
> You're doing really well, I did some trivial comments, that again, because I'm strict for now, I'd like you to address, but basically is that you forgot to run clang-format -i before committing the changes.
> 
> Can we leave the documentation modifications that you did out from this pull request for now? I'd like to propose a documentation format instead.
> 
> So try for now only push the fixes for the warnings.

Hi Enric,
Thanks for the vote of confidence :)    Still much to learn!!!

Sorry for forgetting the clang-format and putting it in a separate commit later.  I was unsure of how to change history......
 
I am happy remove the documentation modifications.   Not sure how I should change pushed history.   Should I "git revert" the commits (which will add new commits) and then push the new commits that reverse them?
git revert bbc2287  (ozmaps/mapmanager.c)
git revert a62b572 (ozmaps/mapmisc.c)
git revert b137d18 (include/maps.h)

Thanks for your help!!!!
Peter

PeterMacleodThompson committed:
Revert "include: maps.h  make comments readable where clang-format made unreadable"

This reverts commit b137d1885731abf0f171998fde07f9206ef462de.

change branch history
removal of documentation modifications ( 1 of 3)
b137d18, ac2b572, bbc2287,

PeterMacleodThompson committed:
Revert "ozmaps: mapmisc.c: add FIXME comment re long/lat structure"

This reverts commit ac2b5723903010243d9084766cea54e16c42509b.

change branch history
removal of documentation modifications ( 2 o 3)
b137d18, ac2b572, bbc2887,

PeterMacleodThompson committed:
Revert "ozmaps: mapmanager.c: clang-format changes to mapmanager.c"

This reverts commit 485d33ff633d53dddafd36cb8b61d7cb7ad1ba4e.
change branch history
Required for removal of
bbc2887-"make comments readable where clang-format made unreadable"
(see next commit)

I could not revert bbc2887 by itself because it gives message
"error: could not revert... after resolving conflicts..."
After revert 485d33f, I could revert bbc2887.
Only other option was to rebase and I see warnings
 "do not rebase if you have already pushed"

PeterMacleodThompson committed:
Revert "ozmaps: mapmanager.c: make comments readable where clang-format made unreadable"

This reverts commit bbc2887384248f8acc48a57ebe805f40a0eea192.
change branch history
removal of documentation modifications ( 3 of 3)
b137d18, ac2b572, bbc2887,

See revert 485d33f-"clang-format changes to mapmanager.c"
to see what a convoluted process this was!!

PeterMacleodThompson committed:
Revert "ozdisplay: Remove DEBUG define displaymap.c, SDL2init.c SDL2misc.c"

This reverts commit ee777dea132b0ef0ccf061c24a61b972903d2e23.
change branch history
split ee777de- Remove DEBUG define displaymap.c, SDL2init.c SDL2misc.c
  - keep DEBUG define removal portion of this commit
  - remove documentation modification portion of this commit
This is done by removing this commit ee777de and creating a
new commit manually to re-apply define DEBUG removal
 (see next commit with same name)

PeterMacleodThompson committed:
ozdisplay: (REPEAT) Remove DEBUG define displaymap.c, SDL2init.c SDL2misc.c

change branch history
manual repeat of "DEBUG removal" portion of commit ee777de
which was removed by previous commit (see previous commit)
Again: #define DEBUG statement was removed from
displaymap.c
SDL2init.c
SDL2misc.c
A #ifdef codeblock was removed in SDL2misc.c
documentation modification portion of commit ee777de is omitted.

Signed-off-by: Peter Macleod Thompson <peter.macleod.thompson@gmail.com>

PeterMacleodThompson committed:
ozmaps: Clang-format "forgive me Father, for I have sinned again"

2 files: mapmanager.c and mapmisc.c were found changed in a
system wide recheck of clang-format
This commit ensures all src is now clang-format-6.0 compatible

Signed-off-by: Peter Macleod Thompson <peter.macleod.thompson@gmail.com>

PeterMacleodThompson commented:
Hi Enric,
The documentation modifications have been removed via 7 commits.
This involved  5 reverts, and 2 adds.  
3 reverts removed pure documentation commits, b137d18, ac2b572, bbc2887
1 revert was a clang-format required to resolve a git conflict: 485d33f conflicted with bbc2887
1 revert was a split removal: part was kept and part was deleted, ee777de
1 add was the clang format replacing commit 485d33f 
1 add was a   ee777de repeat but omitting documentation portion 

Peter

eballetbo commented:
Merged into mainline. Well done!

PeterMacleodThompson commented:
Excellent - thanks
Now onto bigger and better things!! :)


On Thu, Feb 14, 2019 at 3:10 PM Enric Balletbo i Serra <
notifications@github.com> wrote:

> Merged into mainline. Well done!
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/PeterMacleodThompson/hikingGPS/pull/2#issuecomment-463821046>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARf2xrRw6dKA4Xk4idf2x4cPi4TN6nDUks5vNd7bgaJpZM4arKAB>
> .
>

PeterMacleodThompson commented:
Hi Enric,
Should we delete the old for-next branch and restart a new for-next branch pointing to current HEAD?    Or do we continue to extend the old existing for-next forever?   

Whatever you think is best is fine with me.    On my local,  I was planning to delete all "accepted" for-next branches so my new for-next is clean. 

Peter
